src/game.cc

```diff
@@ -1404,125 +1481,129 @@ static void showSplash()
         snprintf(path, sizeof(path), "art\\splash\\");
     }
 
-    File* stream;
-    for (int index = 0; index < SPLASH_COUNT; index++) {
+    // open a splash file, cycle through SPLASH_COUNT entries
+    File* stream = nullptr;
+    for (int attempt = 0; attempt < SPLASH_COUNT; attempt++) {
         char filePath[64];
-        snprintf(filePath, sizeof(filePath), "%ssplash%d.rix", path, splash);
+        snprintf(filePath, sizeof(filePath), "%ssplash%d.rix", path, splashIndex);
         stream = fileOpen(filePath, "rb");
         if (stream != nullptr) {
             break;
         }
 
-        splash++;
-
-        if (splash >= SPLASH_COUNT) {
-            splash = 0;
+        splashIndex++;
+        if (splashIndex >= SPLASH_COUNT) {
+            splashIndex = 0;
         }
     }
 
     if (stream == nullptr) {
         return;
     }
 
+    // Allocate memory for the palette
     unsigned char* palette = reinterpret_cast<unsigned char*>(internal_malloc(768));
     if (palette == nullptr) {
         fileClose(stream);
         return;
     }
 
+    // Verify RIX image version (needed?)
     int version;
     fileReadInt32(stream, &version);
     if (version != 'RIX3') {
         fileClose(stream);
+        internal_free(palette);
         return;
     }
 
-    short width;
-    fileRead(&width, sizeof(width), 1, stream);
-
-    short height;
-    fileRead(&height, sizeof(height), 1, stream);
+    // Read image dimensions
+    short splashWidth, splashHeight;
+    fileRead(&splashWidth, sizeof(splashWidth), 1, stream);
+    fileRead(&splashHeight, sizeof(splashHeight), 1, stream);
 
-    unsigned char* data = reinterpret_cast<unsigned char*>(internal_malloc(width * height));
-    if (data == nullptr) {
+    // Allocate memory for the image pixel data
+    unsigned char* splashData = reinterpret_cast<unsigned char*>(internal_malloc(splashWidth * splashHeight));
+    if (splashData == nullptr) {
         internal_free(palette);
         fileClose(stream);
         return;
     }
 
+    // Load palette and image data
     paletteSetEntries(gPaletteBlack);
     fileSeek(stream, 10, SEEK_SET);
     fileRead(palette, 1, 768, stream);
-    fileRead(data, 1, width * height, stream);
+    fileRead(splashData, 1, splashWidth * splashHeight, stream);
     fileClose(stream);
 
-    // Fix of wrong Palette, without it this makes background bright
-    // Basically just swapping first and last colors, this problem presented ONLY in F2, F1 has right palette in every splash
-    memcpy(palette + (255 * 3), palette, 3);
-    memset(palette, 0, 3);
-
-    for (int i = 0; i < width * height; i++) {
-        if (data[i] == 0) {
-            data[i] = 255;
-        } else if (data[i] == 255) {
-            data[i] = 0;
-        }
-    }
-
-    int size = 0;
```

Comment:
Looks like we dropped this code.  What's the thinking there?

Reply:
Have you tested with Fallout 2 though?  The comments imply that it's only needed there